test/lib/mayaUsd/utils/CMakeLists.txt

```diff
@@ -39,3 +43,65 @@ foreach(script ${INTERACTIVE_TEST_SCRIPT_FILES})
     # Add a ctest label to these tests for easy filtering.
     set_property(TEST ${target} APPEND PROPERTY LABELS utils)
 endforeach()
+
+
+# -----------------------------------------------------------------------------
+# C++ unit tests
+# -----------------------------------------------------------------------------
+
+function(add_mayaUsdLibUtils_test TARGET_NAME)
+    add_executable(${TARGET_NAME})
+
+    # -----------------------------------------------------------------------------
+    # sources
+    # -----------------------------------------------------------------------------
+    target_sources(${TARGET_NAME}
+        PRIVATE
+            main.cpp
+            ${ARGN}
+    )
+
+    # -----------------------------------------------------------------------------
+    # compiler configuration
+    # -----------------------------------------------------------------------------
+    mayaUsd_compile_config(${TARGET_NAME})
+
+    target_compile_definitions(${TARGET_NAME}
+        PRIVATE
+            $<$<STREQUAL:${CMAKE_BUILD_TYPE},Debug>:TBB_USE_DEBUG>
+            $<$<STREQUAL:${CMAKE_BUILD_TYPE},Debug>:BOOST_DEBUG_PYTHON>
+            $<$<STREQUAL:${CMAKE_BUILD_TYPE},Debug>:BOOST_LINKING_PYTHON>
+    )
+
+    # -----------------------------------------------------------------------------
+    # link libraries
+    # -----------------------------------------------------------------------------
+    target_link_libraries(${TARGET_NAME}
+        PRIVATE 
+            GTest::GTest
+            mayaUsdUtils
+            ${MAYA_LIBRARIES}
+            mayaUsd
+    )
+
+    # -----------------------------------------------------------------------------
+    # unit tests
+    # -----------------------------------------------------------------------------
+    mayaUsd_add_test(${TARGET_NAME}
+        COMMAND $<TARGET_FILE:${TARGET_NAME}>
+        ENV
+            "LD_LIBRARY_PATH=${ADDITIONAL_LD_LIBRARY_PATH}"
+            "MAYA_LOCATION=${MAYA_LOCATION}"
+    )
+endfunction()
+
+if(IS_WINDOWS)
+    # There are link problems on Linux and OSX with C++ test using USD + Maya,
```

Comment:
I could not get the C++ unit test to link on Linux or run on OSX.

- On Linux 2020, the linker would fail to find OpenSubDiv
- On Linux 2022, execution ofthe test would fail to find libCg.so
- On OSX master, execution would fail to find libOpenMaya.dylib

But all other platforms / versions would work... so I only kept the test on Windows. It is enough to validate during pre-flight.

Reply:
Please assign it to me after it is merged, I will give it a try.